File: orm/src/main/java/com/example/orm/Person.java
package com.example.orm;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Entity
public class Person extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;

    public String name;
    public LocalDate birth;
    public Status status;

    public static Person findByName(String name){
        return find("name", name).firstResult();
    }

    public static List<Person> findAlive(){
        return list("status", Status.Alive);
    }

    public static void deleteStefs(){
        delete("name", "Stef");
    }

    @Override
    public String toString() {
        return "Person{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", birth=" + birth +
                ", status=" + status +
                '}';
    }
}

enum Status {
    Alive,
    Dead,
    Zombi
}

File: kotlin-orm/src/main/kotlin/com/example/orm/Person.kt
package com.example.orm

import io.quarkus.hibernate.orm.panache.PanacheEntityBase
import java.time.LocalDate
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class Person(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int?=0,
    var name: String?="",
    var birth: LocalDate?= LocalDate.now(),
    var status: Status?=Status.Alive

): PanacheEntityBase() {
    fun findByName(name: String): Person {
        return find<PanacheEntityBase>("name", name).firstResult()
    }

    fun findAlive(): List<Person> {
        return list<Person>("status", Status.Alive)
    }

    fun deleteStefs() {
        delete("name", "Stef")
    }
}

enum class Status {
    Alive,
    Dead,
    Zombi
}